src/goto-instrument/cover_instrument.h

```diff
@@ -44,7 +44,14 @@ class cover_instrumenter_baset
   {
     Forall_goto_program_instructions(i_it, goto_program)
     {
-      instrument(goto_program, i_it, basic_blocks);
+      // do not instrument if instruction has a source location but is marked as
+      // originating from a java virtual function call removal
+      if(
+        i_it->source_location == source_locationt::nil() ||
+        !i_it->source_location.get_java_removed_virtual_call())
```

Comment:
I think this commit may be in the wrong order as I don't think this function has been introduced yet.